packages/admin/cms-admin/src/generator/future/generateForm/generateFormLayout.ts

```diff
@@ -0,0 +1,53 @@
+import { FormConfig, FormLayoutConfig } from "../generator";
+import { Imports } from "../utils/generateImportsCode";
+
+export function generateFormLayout(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    config: FormLayoutConfig<any>,
+    fieldsCode: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    formConfig: FormConfig<any>,
+) {
+    const gqlType = formConfig.gqlType;
+    const instanceGqlType = gqlType[0].toLowerCase() + gqlType.substring(1);
+
+    const imports: Imports = [];
+    let code = "";
+    const formFragmentFields: string[] = [];
+
+    if (config.type === "fieldSet") {
+        imports.push({ name: "FieldSet", importPath: "@comet/admin" });
+        const supportPlaceholder = config.supportText?.includes("{");
+        if (supportPlaceholder) {
+            imports.push({ name: "FormSpy", importPath: "react-final-form" });
+        }
+        code = `
+        <FieldSet
+            collapsible={${String(config.collapsible)}}
+            initiallyExpanded={${String(config.initiallyExpanded)}}
+            title={<FormattedMessage id="${instanceGqlType}.${config.title.replace(/ /g, "")}.title" defaultMessage="${config.title}" />}
```

Comment:
maybe we should add a name property, using title as id seems wrong. (that might be non-english)

Reply:
I wanted to reserve "name" in case we need to support saving the decision state into a separate column. (e.g. conditional fields are not organized in an json-object and therefore it would be easier in the api or any cronjob to check a single "useAlternativeAddress"-field instead of all fields)

so how about "translationIdPrefix" (which becomes optional as soon as we support the name-config for enabled-column)?

Reply:
or formattedMessageIdPrefix or fmIdPrefix? or do you have any other idea?

Reply:
if we add later saving the state, we can perfectly use a name property that we add now.

Reply:
to my understanding the name-property needs to be optional, because the state wont always be saved in an extra column. except we handle saving the state with a "saveStateViaName"-config option, introducing name field first would require an incompatible change later. but probably the "saveStateViaName"-config option is already the better solution, because typing will be easier (two conditional required string fields vs one required string and a defaulted boolean)

Reply:
sorry, I mixed something. fieldSet is only for styling/layout and not for conditional fields (unlike ext-js :-D), so there will probably never be a field handling the initiallyExpanded-state...

regarding the name field: we could just type `name` differently for LayoutFieldConfig than for FormFieldConfig. So I'll add the name field and handle translation-id with it. Because the only "problem" for me is that I'm associating `name` in context of a FieldConfig to be a valid field, but that's only true for FormFieldConfigs...